### PR TITLE
SDAP-72

### PR DESCRIPTION
**Pull request by quintinali**
Fixed bugs in ESDriver.java

(1)"InterruptedException" should not be ignored
(2)Use try-with-resources or close this "ESTransportClient" in a
"finally" clause.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: core/src/main/java/org/apache/sdap/mudrod/driver/ESDriver.java [hunk 1]
```diff
@@ -473,12 +474,20 @@ protected Client makeClient(Properties props) throws IOException {
 
     Client client = null;
 
-    // Prefer TransportClient
+    // Prefer to TransportClient
```
Comment (lewismc) [inline core/src/main/java/org/apache/sdap/mudrod/driver/ESDriver.java]: This comment doesn't make sense.

File: core/src/main/java/org/apache/sdap/mudrod/driver/ESDriver.java [hunk 2]
```diff
@@ -473,12 +474,20 @@ protected Client makeClient(Properties props) throws IOException {
...
     if (hosts != null && port > 1) {
-      TransportClient transportClient = new ESTransportClient(settings);
-      for (String host : hosts)
-        transportClient.addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(host), port));
-      client = transportClient;
+      TransportClient transportClient = null;
+      try {
```
Comment (lewismc) [inline core/src/main/java/org/apache/sdap/mudrod/driver/ESDriver.java]: Use a [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) statement such that the client is closed.

File: core/src/main/java/org/apache/sdap/mudrod/driver/ESDriver.java [hunk 3]
```diff
@@ -473,12 +474,20 @@ protected Client makeClient(Properties props) throws IOException {
...
+        transportClient = new ESTransportClient(settings);
+        for (String host : hosts){
+          transportClient.addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(host), port));
+        }
+        client = transportClient;
+      } catch (Exception e) {
+        
+      } finally {
```
Comment (lewismc) [inline core/src/main/java/org/apache/sdap/mudrod/driver/ESDriver.java]: You've added the finally statement, but you have not closed the client. Just use the try-with-resources statement OK.

**Review by lewismc — CHANGES_REQUESTED** [review]
Please address comments @quintinali

File: core/src/main/java/org/apache/sdap/mudrod/driver/ESDriver.java (hunk 2 above)
Comment (quintinali) [reply core/src/main/java/org/apache/sdap/mudrod/driver/ESDriver.java]: @lewismc, I don't want to close the client in this function. If I change the code to 

if (hosts != null && port > 1) {  
      try (TransportClient transportClient = new ESTransportClient(settings)) {
        for (String host : hosts){
          transportClient.addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(host), port));
        }
        client = transportClient;
      }
    }

An error occurs as following:
java.lang.IllegalStateException: transport client is closed
	at org.elasticsearch.client.transport.TransportClientNodesService.execute(TransportClientNodesService.java:240)
	at org.elasticsearch.client.transport.TransportProxyClient.execute(TransportProxyClient.java:59)
	at org.elasticsearch.client.transport.TransportClient.doExecute(TransportClient.java:356)
	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:403)
	at org.elasticsearch.client.support.AbstractClient$IndicesAdmin.execute(AbstractClient.java:1226)
	at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:80)
	at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:54)
	at org.apache.sdap.mudrod.driver.ESDriver.putMapping(ESDriver.java:142)
	at org.apache.sdap.mudrod.discoveryengine.MudrodAbstract.initMudrod(MudrodAbstract.java:84)
	at org.apache.sdap.mudrod.discoveryengine.MudrodAbstract.<init>(MudrodAbstract.java:56)
	at org.apache.sdap.mudrod.discoveryengine.DiscoveryEngineAbstract.<init>(DiscoveryEngineAbstract.java:29)
	at org.apache.sdap.mudrod.discoveryengine.WeblogDiscoveryEngine.<init>(WeblogDiscoveryEngine.java:50)
	at org.apache.sdap.mudrod.main.MudrodEngine.startFullIngest(MudrodEngine.java:242)
	at org.apache.sdap.mudrod.main.MudrodEngine.main(MudrodEngine.java:388)

**Comment by lewismc** [comment]
Are you still having issues with the try-with-resources?

**Comment by quintinali** [comment]
@lewismc Yes, I still have issues with the try-with-resources. 

If I create ESTransportClient using try-with-resources as follow, an error occurs.
```
    if (hosts != null && port > 1) {  
      try (TransportClient transportClient = new ESTransportClient(settings)) {
        for (String host : hosts){
          transportClient.addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(host), port));
        }
        client = transportClient;
      }
    } else if (clusterName != null) {
      node = new Node(settings);
      client = node.client();
    }
```
2018-05-17 22:17:02,276 ERROR main.MudrodEngine (MudrodEngine.java:main(398)) - Error whilst parsing command line.
java.lang.IllegalStateException: transport client is closed
	at org.elasticsearch.client.transport.TransportClientNodesService.execute(TransportClientNodesService.java:240)
	at org.elasticsearch.client.transport.TransportProxyClient.execute(TransportProxyClient.java:59)
	at org.elasticsearch.client.transport.TransportClient.doExecute(TransportClient.java:356)
	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:403)
	at org.elasticsearch.client.support.AbstractClient$IndicesAdmin.execute(AbstractClient.java:1226)
	at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:80)
	at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:54)
	at org.apache.sdap.mudrod.driver.ESDriver.putMapping(ESDriver.java:142)
	at org.apache.sdap.mudrod.discoveryengine.MudrodAbstract.initMudrod(MudrodAbstract.java:84)
	at org.apache.sdap.mudrod.discoveryengine.MudrodAbstract.<init>(MudrodAbstract.java:56)
	at org.apache.sdap.mudrod.discoveryengine.DiscoveryEngineAbstract.<init>(DiscoveryEngineAbstract.java:29)
	at org.apache.sdap.mudrod.discoveryengine.WeblogDiscoveryEngine.<init>(WeblogDiscoveryEngine.java:50)
	at org.apache.sdap.mudrod.main.MudrodEngine.startFullIngest(MudrodEngine.java:242)
	at org.apache.sdap.mudrod.main.MudrodEngine.main(MudrodEngine.java:388)

**Review by lewismc — APPROVED** [review]

**Pull request merged**